packages/sheets-ui/src/commands/commands/add-worksheet-merge.command.ts

```diff
@@ -94,14 +95,27 @@ export const AddWorksheetMergeCommand: ICommand = {
             subUnitId,
             ranges,
         };
-        redoMutations.push({ id: RemoveWorksheetMergeMutation.id, params: removeMergeMutationParams });
+
+        const hasIntersect = worksheet.getMergeData().some((merge) => {
+            return ranges.some((range) => {
```

Comment:
You can use arrow function expression to make the code more compact and easier to read:

```ts
const hasIntersection = worksheet.getMergeData().some((merge) => ranges.some((range) => Rectangle.intersects(range, merge)))
```